api/v1alpha1/hostedcluster_types.go

```diff
@@ -54,6 +54,30 @@ type Release struct {
 	Image string `json:"image"`
 }
 
+type HostedClusterCondition struct {
```

Comment:
Is there a reason to not use core condition types?

Reply:
I'm glad you brought this up, I'd been meaning to ask and keep forgetting... can we use a shared condition type for all our conditions on these API resources? If so, which? And are there existing libs for working with them, and if so where, etc. I have no reason to prefer distinct types, especially since we have no generics in the language. @csrwng @sjenning

Reply:
fwiw In nodePools we are using core types and library for conditions atm.